pageserver/src/tenant/storage_layer/layer.rs

```diff
@@ -880,23 +880,18 @@ impl LayerInner {
     ) -> Result<heavier_once_cell::InitPermit, DownloadError> {
         debug_assert_current_span_has_tenant_and_timeline_id();
 
-        let task_name = format!("download layer {}", self);
-
         let (tx, rx) = tokio::sync::oneshot::channel();
 
-        // this is sadly needed because of task_mgr::shutdown_tasks, otherwise we cannot
-        // block tenant::mgr::remove_tenant_from_memory.
-
         let this: Arc<Self> = self.clone();
 
-        crate::task_mgr::spawn(
-            &tokio::runtime::Handle::current(),
-            crate::task_mgr::TaskKind::RemoteDownloadTask,
```

Comment:
We can remove that TaskKind from task_mgr.rs

Reply:
Good idea to clean it up as well. Weird no warning again...?

Reply:
No warning because it is `pub`, used by requestcontext task_mgr::spawn from `pageserver/src/bin/pageserver.rs`.

Reply:
https://github.com/neondatabase/neon/pull/7051/commits/84fff41e2422c77fb63a0319017795a4a5cbefd4